lib/queuePopulator/QueuePopulatorExtension.js

```diff
@@ -0,0 +1,44 @@
+const { EventEmitter } = require('events');
+
+class QueuePopulatorExtension extends EventEmitter {
+    /**
+     * @constructor
+     * @param {Object} params - constructor params
+     * @param {Object} params.config - extension-specific
+     *   configuration object
+     * @param {Logger} params.logger - logger object
+     */
+    constructor(params) {
+        super();
+        this.log = params.logger;
+        this._batch = null;
+    }
+
+    setBatch(batch) {
+        this._batch = batch;
+    }
+
+    /**
+     * Prepare a kafka entry for publishing
+     *
+     * This function is meant to be used in subclasses' filter()
```

Comment:
I think you mean _onEntry()?

Reply:
It's a left-over, actually I mean that it's meant to be called after `entry` events are received.